### DISCUSSION TITLE
Janssen LDAP federation with Jumpcloud

### DISCUSSION THREAD

**Discussion by 0786ShivaniBansal** [JanssenProject/jans#6001] [Q&A] [answered]
Hii Team we have started LDAP Federation with Jumpcloud by following this documentation
https://docs.jans.io/v1.0.17.nightly/admin/link/

I'm attaching the configuration screenshots please go through it 
after adding source backend LDAP server and using this command 

```
/opt/opendj/bin/ldapsearch -H https://console.jumpcloud.com:636 -D "uid=ShivaniBansal,ou=Users,o=62c14e7eb411205b3ce106f8,dc=jumpclo
ud,dc=com" -w xxxx -b "o=62c14e7eb411205b3ce106f8,dc=jumpcloud,dc=com" dn | grep "dn:" | wc -l
```
we are getting o/p=0

While go through jans-link.log this is the error

```
08-09-2023 10:18:24.056 INFO  [ForkJoinPool.commonPool-worker-5] jans.link.timer.CacheRefreshTimer CacheRefreshTimer.java:208- Exception happened while executing cache refresh synchronizationio.jans.orm.exception.operation.ConfigurationException: Failed to create LDAP connection pool! Result code: 'null'
08-09-2023 10:18:24.056 INFO  [ForkJoinPool.commonPool-worker-5] jans.link.timer.CacheRefreshTimer CacheRefreshTimer.java:208- Exception happened while executing cache refresh synchronizationio.jans.orm.exception.operation.ConfigurationException: Failed to create LDAP connection pool! Result code: 'null'
```

please verify and reply as soon as possible

Thanks
![configuration](https://github.com/JanssenProject/jans/assets/95275127/f7910e2b-a12d-4f19-b591-6fc16e45bc95)
![customerbackendkeyattribute](https://github.com/JanssenProject/jans/assets/95275127/6d5ad8e6-d5b9-48f8-89eb-0447db8adbfe)
![defaultinumserver](https://github.com/JanssenProject/jans/assets/95275127/76fdafb0-20f2-4fcd-9ef5-73a704a7ff6b)
![sourcebackendldapserver](https://github.com/JanssenProject/jans/assets/95275127/981fa5ff-c2d4-4390-8fac-22713af3ec43)
![result](https://github.com/JanssenProject/jans/assets/95275127/151f325a-c1e6-4019-99ec-98f87e0d02be)

**Comment by 0786ShivaniBansal** [comment]
Hii Team any update?

**Comment by mzico** [comment]
```
 Failed to create LDAP connection pool! Result code: 'null'
 ```
 
 Seems like something is off with network connectivity.

**Comment by nynymike** [comment]
```
/opt/opendj/bin/ldapsearch -H https://console.jumpcloud.com:636
```

I've never seen this work. Can you try ldapsearch with  

```
`-h host -p port -X -Z \
  -D "uid=ShivaniBansal,ou=Users,o=62c14e7eb411205b3ce106f8,dc=jumpcloud,dc=com"  \
  -b "dc=jumpcloud,dc=com"  \ 
  -s base \ 
  "objectclass=*" 
```

BTW, this should prompt you interactively for the password, or use `-j` and put the pw in a file (don't put the root pw in the shell... you don't want it in the history). 

If you can't run ldapsearch from the Jans Auth Server, then Link will never work.

**Comment by mmrraju** [comment, marked as answer]
Hello @0786ShivaniBansal 

**Failed to create LDAP connection pool! Result code: 'null'**

You may check your source LDAP server connection is establish or not just simply using Jans TUI

Go to `Jans-Auth>> Auth>> Add Source LDAP server` and Test 

Then follow the [Jans-Link](https://docs.jans.io/v1.0.17.nightly/admin/link/)
